test/test_scandir.py

```diff
@@ -250,6 +250,33 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_walk_unicode_handling(self):
+        if sys.platform == 'win32':
+            return self.skipTest("Not windows")
```

Comment:
Hum no, it's not good to skip the test on some platforms. You must test the type on all platforms.